app/src/components/code/JSONEditor.tsx

```diff
@@ -24,24 +24,35 @@ export type JSONEditorProps = Omit<
    * JSON Schema to use for validation, if provided will enable JSON Schema validation with tooltips in the editor
    */
   jsonSchema?: JSONSchema7;
+  /**
+   * If true, will not lint the content if it is empty
+   */
+  optionalLint?: boolean;
 };
 
 export function JSONEditor(props: JSONEditorProps) {
   const { theme } = useTheme();
-  const { jsonSchema, ...restProps } = props;
+  const { jsonSchema, optionalLint, ...restProps } = props;
   const codeMirrorTheme = theme === "light" ? githubLight : nord;
   // Force a refresh of the editor when the jsonSchema changes
   // Code mirror does not automatically refresh when the extensions change
   const [schemaKey, setSchemaKey] = React.useState(0);
   useEffect(() => {
     setSchemaKey((prev) => prev + 1);
   }, [jsonSchema]);
+  const contentLength = props.value?.length;
+  // When optionalLint is false, this value will always be true
+  // This prevents the extensions useMemo from needlessly re-running when optionalLint is false
+  const hasContent = !optionalLint || (contentLength && contentLength > 0);
```

Comment:
can we change this to be something like shouldLint? and then just use it below as well?

After reading a few times i'm kinda starting to grok this but it feels confusing, why did you need to add this? I think hinging the extensions based on the content is going to be rough

Reply:
The rename makes sense. I introduced this because otherwise, empty JSONEditor fields trigger a lint error.

Empty JSONEditor fields are valid in the invocation params form so we should only enable linting when there is content in that case.

The optionalLint prop was added so that this new functionality only takes place when optionalLint is true, otherwise it will keep linting at all times just like the original behavior.

For what it is worth, we only register extensions when content length goes from 0 -> n, or n -> 0, not on every keystroke.

Reply:
Renamed and simplified logic

Reply:
gotcha, that makes sense, i think controlling when the useMemo runs is probably good the names were just a bit confusing update looks good